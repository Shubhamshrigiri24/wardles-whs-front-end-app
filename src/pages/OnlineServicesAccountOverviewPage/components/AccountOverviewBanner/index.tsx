import React, { FC } from "react";
import { Container, Paper, makeStyles, Theme, Box } from "@material-ui/core";
import { wellColors, Typography } from "@welldigital/components";

const useStyles = makeStyles((theme: Theme) => ({
  banner: {
    backgroundColor: wellColors.greyscale[100],
  },
  bannerTitle: {
    textTransform: "capitalize",
    fontSize: "34px",
    lineHeight: "125%",
    color: wellColors.elixir[900],
    [theme.breakpoints.up("md")]: {
      fontSize: "48px",
    },
  },
  bannerText: {
    fontWeight: 500,
    color: wellColors.elixir[900],
  },
}));

const AccountOverviewBanner: FC<{ orderType: string }> = ({ orderType }) => {
  const classes = useStyles();
  return (
    <Paper classes={{ root: classes.banner }} elevation={0}>
      <Container maxWidth={"lg"}>
        <Box py={6}>
          <Typography
            classes={{ root: classes.bannerTitle }}
            variant={"h1"}
            spacingAfter={3}
            data-testid={"accountOverviewBanner/title"}
          >
            {orderType}
          </Typography>
          <Box maxWidth={520}>
            <Typography
              data-testid={"accountOverviewBanner/content"}
              variant={"h4"}
              className={classes.bannerText}
            >
              Check the dates and status of your {orderType}.
            </Typography>
          </Box>
        </Box>
      </Container>
    </Paper>
  );
};

export default AccountOverviewBanner;
