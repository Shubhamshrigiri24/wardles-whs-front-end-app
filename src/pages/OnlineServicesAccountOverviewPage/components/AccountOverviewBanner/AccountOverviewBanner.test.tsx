import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import AccountOverviewBanner from "./index";

describe("<AccountOverviewBanner />", () => {
  it("should match snapshot", () => {
    const { container } = render(<AccountOverviewBanner />);
    expect(container).toMatchSnapshot();
  });

  it("should render title and content", () => {
    const { getByTestId } = render(<AccountOverviewBanner />);
    const title = getByTestId("accountOverviewBanner/title");
    const content = getByTestId("accountOverviewBanner/content");
    expect(title).toBeInTheDocument();
    expect(content).toBeInTheDocument();
  });
});
