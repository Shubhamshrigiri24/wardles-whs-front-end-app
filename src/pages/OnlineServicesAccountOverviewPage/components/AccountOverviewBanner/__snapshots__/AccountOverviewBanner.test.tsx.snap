// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountOverviewBanner /> should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-banner-1 MuiPaper-elevation0 MuiPaper-rounded"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <h1
          class="MuiTypography-root makeStyles-bannerTitle-2 MuiTypography-h1"
          data-testid="accountOverviewBanner/title"
        />
        <div
          class="makeStyles-spacing3-11 makeStyles-spacing3Responsive-12"
          data-testid="spacing/root"
        />
        <div
          class="MuiBox-root MuiBox-root-31"
        >
          <h4
            class="MuiTypography-root makeStyles-bannerText-3 MuiTypography-h4"
            data-testid="accountOverviewBanner/content"
          >
            Check the dates and status of your 
            .
          </h4>
          <div
            class="makeStyles-spacing3-11 makeStyles-spacing3Responsive-12"
            data-testid="spacing/root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
